### ISSUE TITLE
graph drawing improvements

### ISSUE THREAD

**Issue by dkochmanski** [McCLIM/McCLIM#975]
- arcs are drawn in line with x/y axis
- merge-duplicated-nodes new option which does not merge, but doesn't expand duplicates either
- [when we have expandable output records] toggle details

![obraz](https://user-images.githubusercontent.com/6871201/75465430-ea53b480-5988-11ea-9ae7-b813caba9c64.png)

**Comment by john-a-carroll** [comment]
I'm a heavy user of `format-graph-from-roots` and am very happy to see enhancements to it. But I feel that "arcs are drawn in line with x/y axis" should be an **option** and **not** the default. It works for some kinds of trees, and indeed it is a conventional arrangement for organization charts. But it's not conventional for trees that arise in many areas of computer science, e.g. parse trees (see https://en.wikipedia.org/wiki/Parse_tree). And this kind of output would be very confusing for DAGs (see the examples at https://en.wikipedia.org/wiki/Directed_acyclic_graph).

**Comment by scymtym** [comment]
I experimented with something like this a bit. I used the `:arc-drawer` option of `format-graph-from-roots` to allow for different arc drawing styles: https://techfak.de/~jmoringe/mcclim-graph-layout-1.png (the choices at the top are for `:orientation` and `:arc-drawer`).

**Comment by john-a-carroll** [comment]
Here's a small example of one of the kinds of graph output by my application: http://lists.delph-in.net/archives/developers/attachments/20171220/86a6a919/attachment-0003.png . In this case, the customization is mainly to `object-printer`.

**Comment by scymtym** [comment]
@john-a-carroll i think your application would not be affected at all if we provided alternate edge drawing styles as additional functions to supply via `:arc-drawer`.

**Comment by john-a-carroll** [comment]
@scymtym Yes, I agree, this is the correct thing to do. (In fact, at present if `format-graph-from-roots` is called with `:graph-type :digraph` then it uses a different default `:arc-drawer`, which adds arrow heads and draws circular arcs for nodes that point to themselves).
